### PR TITLE
Adding dependabot rules

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
This PR introduces Dependabot rules to keep our dependencies up-to-date and secure. 

Dependabot will now:

Weekly check for updates: It will proactively scan for the latest versions of our dependencies.
Open pull requests: When updates are available, it will create PRs to seamlessly integrate them.
Prioritize security: It'll focus on addressing vulnerabilities and keeping our codebase secure.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+enable-beta-ecosystems: true
+updates:
+- package-ecosystem: docker
```
Comment (songjiaxun) [inline .github/dependabot.yml:4]: Do we also plan to scan go.mod?

Comment (Tulsishah) [reply .github/dependabot.yml:4]: Dependabot can indeed generate alerts for Go projects without an explicit dependabot.yml file. It does this by automatically scanning repositories for supported manifest files, including go.mod.

We are already getting alerts for go.mod file.

**Review by songjiaxun — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (Tulsishah) [reply .github/dependabot.yml:4]: I got to know that writing yml file will disable the default alerts, that's why adding more alerts.

**Review by Tulsishah — COMMENTED** [review]
@sethiay please take a look

**Review by sethiay — COMMENTED** [review]
please update the PR description and title of PR.

**Review by sethiay — APPROVED** [review]

**Pull request merged**